api/test/test_play_asset_pack_manager.gd

```diff
@@ -409,3 +409,324 @@ func assert_remove_pack_signal_is_error(did_succeed : bool, exception : PlayAsse
+	yield(yield_to(test_object, "state_updated", 1), YIELD)
+	assert_signal_emitted(test_object, "state_updated")
+	assert_false(request_object.get_is_completed())
+	assert_asset_pack_state_eq_dict(request_object.get_state(), test_asset_pack_state)
+	
+	# Emit and assert a stream of state_updated signal
+	var updated_state1 = create_mock_asset_pack_state_with_status_and_progress_dict(test_pack_name, \
+		PlayAssetPackManager.AssetPackStatus.DOWNLOADING, 256, 4096)
+	var updated_signal_info1 = FakePackStateInfo.new(updated_state1)
+	mock_plugin.trigger_asset_pack_state_updated_signal(updated_signal_info1)
+	yield(yield_to(test_object, "state_updated", 1), YIELD)
+	updated_signal_info1.thread.wait_to_finish()
+	
+	assert_signal_emitted(test_object, "state_updated")
+	assert_false(request_object.get_is_completed())
+	assert_asset_pack_state_eq_dict(request_object.get_state(), updated_state1)
+	
+	var updated_state2 = create_mock_asset_pack_state_with_status_and_progress_dict(test_pack_name, \
+		PlayAssetPackManager.AssetPackStatus.WAITING_FOR_WIFI, 256, 4096)
+	var updated_signal_info2 = FakePackStateInfo.new(updated_state2)
+	mock_plugin.trigger_asset_pack_state_updated_signal(updated_signal_info2)
+	yield(yield_to(test_object, "state_updated", 1), YIELD)
+	updated_signal_info2.thread.wait_to_finish()
+	
+	assert_signal_emitted(test_object, "state_updated")
+	assert_false(request_object.get_is_completed())
+	assert_asset_pack_state_eq_dict(request_object.get_state(), updated_state2)
+
+	var updated_state3 = create_mock_asset_pack_state_with_status_and_progress_dict(test_pack_name, \
+		PlayAssetPackManager.AssetPackStatus.DOWNLOADING, 2048, 4096)
+	var updated_signal_info3 = FakePackStateInfo.new(updated_state3)
+	mock_plugin.trigger_asset_pack_state_updated_signal(updated_signal_info3)
+	yield(yield_to(test_object, "state_updated", 1), YIELD)
+	updated_signal_info3.thread.wait_to_finish()
+	assert_signal_emitted(test_object, "state_updated")
+	assert_false(request_object.get_is_completed())
+	assert_asset_pack_state_eq_dict(request_object.get_state(), updated_state3)
+	
+	var updated_state4 = create_mock_asset_pack_state_with_status_and_progress_dict(test_pack_name, \
+		PlayAssetPackManager.AssetPackStatus.COMPLETED, 4096, 4096)
+	var updated_signal_info4 = FakePackStateInfo.new(updated_state4)
+	mock_plugin.trigger_asset_pack_state_updated_signal(updated_signal_info4)
+	yield(yield_to(request_object, "state_updated", 1), YIELD)
+	updated_signal_info4.thread.wait_to_finish()
+	
+	assert_signal_emitted(test_object, "state_updated")
+	assert_true(request_object.get_is_completed())
+	assert_asset_pack_state_eq_dict(request_object.get_state(), updated_state4)	
+	
+	# assert signal_captor is as expected
+	var result_params_store = signal_captor.received_params_store
+	var expected_state_list = [test_asset_pack_state, updated_state1, updated_state2, \
+		updated_state3, updated_state4]
+	assert_eq(result_params_store.size(), 5)
```

Comment:
Declare 5 as a separate variable so we can give it semantic meaning: `num_updates` or something.

Reply:
Refactored to use ```	var expected_num_updates = 5 ```